### PR TITLE
add support for internal votes

### PR DESCRIPTION
**Pull request by bayesian**
@criver , @YaoXu

### CODE REVIEW

File: Tree.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ template <class T>
 class PartitionNode : public TreeNode<T> {
  public:
   PartitionNode(int fid, T fv)
-    : fid_(fid), fv_(fv),
+    : fid_(fid), fv_(fv), fvote_(0.0),
```
Comment (criver) [inline Tree.h]: fv_ and fvote_ is the same thing, right. Can we just use one?

Comment (bayesian) [inline Tree.h]: fv_ is the split value.. fvote_ is the "leaf value" if the intermediate node does not split further.

**Comment by bayesian** [comment]
tested on training data, generated model like:

{
  "trees" : [
    {
      "value" : 0.013990319666282362,
      "index" : -1
    },
    {
      "vote" : 5.57184536321531E-7,
      "right" : {
        "vote" : 0.0074322169839211374,
        "right" : {
          "vote" : 0.02272090847828602,
          "right" : {
            "vote" : 0.01676262541279356,
            "right" : {
              "vote" : 0.02702191299876493,
              "right" : {
                "value" : 0.0031828663213962927,
                "index" : -1
              },
              "left" : {
                "value" : 0.00015836272436282507,
                "index" : -1
              },
              "value" : -1,
              "index" : 22
            },

**Comment by criver** [comment]
LGTM.

**Pull request merged**